### ISSUE TITLE
Create an editor setting for selecting which icons get shown in the scene tree view

### ISSUE THREAD

**Issue by Gnumaru** [godotengine/godot-proposals#9951]
### Describe the project you are working on

More than one godot project.

### Describe the problem or limitation you are having in your project

We could have much more usefull icons in the scene tree view, but horizontal space is limited. So we could just let the user select what he wants to be shown there.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

There would be a new section in the editor settings listing all the possible icons shown in the scene tree view and letting the user select what whe wants to be shown there. So if the user wants to show all icons and live with the lack of space, he can do so. If he doesn't want all the clutter, whe can just chose the most essential icons like the script icon and the external scene icon.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

There would be a new section in the editor settings listing all the possible icons shown in the scene tree view. Each icon name would have a checkbox. If the user marks it the icon get's shown. If he unchecks it, the icon is always hidden. Optionally, when hovering the node, a list of all the icons for that node is shown (this could be another proposal).

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I suppose it can. I'm really not familiar with using addons to hack parts of the editor UI that are not meant to be changed, but I have a hunch it could be done with an addon.

### Is there a reason why this should be core and not an add-on in the asset library?

This could allow the creation of several more useful icons for the scene tree, so I think most people would benefit of it being on core than an addon.

**Comment by Calinou** [comment]
If this is a project setting, then this would affect all team members working on the project. However, different team members might have different needs (and screen sizes), so I don't think they should be forced to see the same set of icons.

**Comment by Gnumaru** [comment]
Indeed. Since this would be something to cater to the personal tastes of each user, it would be terrible to enforce this for everyone working on the same project. An editor setting would be much better. I'm gonna change the proposal.

**Comment by passivestar** [comment]
I suggested the same in [this comment](https://github.com/godotengine/godot-proposals/issues/7715#issuecomment-2134035729)
